mk/spksrc.archs.mk

```diff
@@ -55,3 +55,15 @@ DOTNET_UNSUPPORTED_ARCHS += armadaxp
 
 # compatibility with .NET not yet confirmed:
 # alpine4k armada375 armada38x comcerto2k
+
+# Exclusions for dotnet 6.0 core apps
+ifeq ($(strip $(DOTNET_CORE_ARCHS)),1)
+    UNSUPPORTED_ARCHS = $(PPC_ARCHS) $(ARMv5_ARCHS) $(ARMv7L_ARCHS) $(i686_ARCHS) armada370
+    UNSUPPORTED_ARCHS_TCVERSION = armv7-6.1
+endif
+
+# Exclusions for dotnet 6.0 servarr apps (except x86)
+ifeq ($(strip $(DOTNET_SERVARR_ARCHS)),1)
+    UNSUPPORTED_ARCHS = $(PPC_ARCHS) $(ARMv5_ARCHS) $(ARMv7L_ARCHS) armada370
+    UNSUPPORTED_ARCHS_TCVERSION = armv7-6.1
+endif
```

Comment:
I like the `UNSUPPORTED_ARCHS_TCVERSION` which gives more flexibility, and actually didn't recall it existed :shrug: .  Although I'm thinking this could just be merged by allowing `UNSUPPORTED_ARCHS` to support plain archs calls along with more specific ones (e.g. `<arch>-<version>`).  This would be much simpler to support by adapting `spksrc.pre-check.mk` in consequence.  Perhaps for another time?

As for adding specific unsupported package caracteristics in a generic framework makefile, I'm not too keen on this.  I would really favor using package specific makefiles.  Now I just noticed that there already is other package specific entries in there.  At the very least, if not feasible to push them down to the package makefiles, please move them out from `arch.mk` by including a sub-file specific for package exceptions.

Reply:
hey @th0ma7, sorry I was a bit too eager to get this merged as I had a number of package updates waiting on this. As for the use of `UNSUPPORTED_ARCHS_TCVERSION`, I only used this shorten the number of lines in the code. I would have preferred to use something like `ifeq ($(call version_lt, ${TCVERSION}, 7.0),1)` to determine if we are building for DSM 6.x rather than being specific to DSM 6.1. I opted to leave it as-is because it's not likely to change (since we will probably not have a DSM 6.3 and if we do change the build to output for DSM 6.2 specifically, the arch file can be revisited then).

Regarding your comment for generic framework makefiles, I did have these included previously for a number of `*arr` apps as an attempt to correct the use of the `DOTNET_UNSUPPORTED_ARCHS` variable included as part of #5383. The dotnet compatibility was not only arch specific but also DSM version specific so my intent over time is to remove references to `DOTNET_UNSUPPORTED_ARCHS` and remove this from the framework.

I'm not familiar with sub-file specific for package exceptions, perhaps you can share more information on this?

Reply:
That's quite easy, you simply create for instance a spksrc.arch-exceptions.mk and use include from arch.mk.

This way arch stays clean and exceptions are kept aside.

Reply:
I extracted spksrc.archs.mk from spksrc.common.mk (about 2 years ago)
and I like that every arch specific configuration is placed in spksrc.archs.mk

IMHO it is not needed to introduce an additional level of includes by extraction of definitions to spksrc.arch-exceptions.mk.